### PR TITLE
Fix sitemap

### LINKED ISSUE #698: Invalid links in sitemap
Discovered in Search Console:

https://almanac.httparchive.org/en/2019/table_of_contents should be https://almanac.httparchive.org/en/2019/table-of-contents

https://almanac.httparchive.org/en/2019/accessibility_statement should be https://almanac.httparchive.org/en/accessibility-statement

### PR DESCRIPTION
**Pull request by bazzadp**
Fixes #698

### CODE REVIEW

File: src/templates/sitemap.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
     </url>
 
     <url>
-        <loc>https://almanac.httparchive.org/en/2019/accessibility_statement</loc>
+        <loc>https://almanac.httparchive.org/en/2019/accessibility-statement</loc>
```
Comment (rviscomi) [inline src/templates/sitemap.xml]:
```suggestion
        <loc>https://almanac.httparchive.org/en/accessibility-statement</loc>
```

File: src/templates/sitemap.xml [hunk 2]
```diff
@@ -117,7 +117,7 @@
     </url>
 
     <url>
-        <loc>https://almanac.httparchive.org/en/2019/table_of_contents</loc>
+        <loc>https://almanac.httparchive.org/en/2019/table-of_contents</loc>
```
Comment (rviscomi) [inline src/templates/sitemap.xml]:
```suggestion
        <loc>https://almanac.httparchive.org/en/2019/table-of-contents</loc>
```

File: src/tools/generate/generate_sitemap.js [hunk 3]
```diff
@@ -91,14 +91,14 @@ const set_min_date = (date) => {
...
 };
 
-const strip_html_extension = (url) => {
+const convert_file_name = (url) => {
   if ( url.substr(url.length - 10) == "index.html" ) {
     return url.substr(0, url.length - 10);
   };
   if ( url.substr(url.length - 5) == ".html" ) {
-    return url.substr(0, url.length - 5);
+    return url.substr(0, url.length - 5).replace('_','-');
```
Comment (rviscomi) [inline src/tools/generate/generate_sitemap.js]: Does this do a global replace? TOC page still has an underscore. Same for next `replace` call.

Comment (bazzadp) [reply src/tools/generate/generate_sitemap.js]: Nope it doesn't. Stupid language. Fixed with a regex. Bout time we had [replaceAll](https://v8.dev/features/string-replaceall)!

File: src/templates/sitemap.xml (hunk 1 above)
Comment (rviscomi) [inline src/templates/sitemap.xml]: I think this still has an issue pointing to 2019 when the live URL is https://almanac.httparchive.org/en/accessibility-statement

File: src/tools/generate/generate_sitemap.js [hunk 4]
```diff
@@ -91,14 +91,21 @@ const set_min_date = (date) => {
...
 };
 
-const strip_html_extension = (url) => {
+const convert_file_name = (url) => {
   if ( url.substr(url.length - 10) == "index.html" ) {
     return url.substr(0, url.length - 10);
   };
   if ( url.substr(url.length - 5) == ".html" ) {
-    return url.substr(0, url.length - 5);
+    if ( url.substr(url.length - 28) == "accessibility_statement.html" ) {
```
Comment (rviscomi) [inline src/tools/generate/generate_sitemap.js]: Does this simplification still work?
```suggestion
    if ( url.endsWith("accessibility_statement.html") ) {
```

File: src/templates/sitemap.xml [hunk 5]
```diff
@@ -11,11 +11,6 @@
         <lastmod>2019-11-23</lastmod>
     </url>
 
-    <url>
```
Comment (rviscomi) [inline src/templates/sitemap.xml]: Looks like the entry was removed entirely?

File: src/tools/generate/generate_sitemap.js (hunk 4 above)
Comment (bazzadp) [reply src/tools/generate/generate_sitemap.js]: Yes it does. Clever JavaScript! 😀

File: src/templates/sitemap.xml (hunk 5 above)
Comment (bazzadp) [reply src/templates/sitemap.xml]: No it wasn't. It was just moved to line 125 as it lists them in order and removing year changed the order.

**Pull request merged**